demos/dqx_llm_demo.py

```diff
@@ -0,0 +1,316 @@
+print(f"LLM PK Detection: {config.llm_config.enable_pk_detection}")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Method 2: Options-based enablement
+
+# COMMAND ----------
+
+ws = WorkspaceClient()
+profiler = DQProfiler(ws)
+
+# Enable via options parameter
+summary_stats, dq_rules = profiler.profile_table(
+    "catalog.schema.table", 
+    options={"llm": True}  # Simple LLM enablement
+)
+print("✅ LLM-based profiling enabled!")
+
+# Check if primary key was detected
+if "llm_primary_key_detection" in summary_stats:
+    pk_info = summary_stats["llm_primary_key_detection"]
+    print(f"Detected PK: {pk_info['detected_columns']}")
+    print(f"Confidence: {pk_info['confidence']}")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Method 3: Direct detection method
+
+# COMMAND ----------
+
+# Direct LLM-based primary key detection
+result = profiler.detect_primary_keys_with_llm(
+    table="customers",
+    llm=True,  # Explicit LLM enablement required
+    options={
+        "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct"
+    }
+)
+
+if result and result.get("success", False):
+    print(f"✅ Detected PK: {result['primary_key_columns']}")
+    print(f"Confidence: {result['confidence']}")
+    print(f"Reasoning: {result['reasoning']}")
+else:
+    print("❌ Primary key detection failed or returned no results")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Rules Generation with is_unique
+# MAGIC 
+# MAGIC Once primary keys are detected via LLM, DQX can automatically generate `is_unique` data quality rules to validate the uniqueness of those columns.
+
+# COMMAND ----------
+
+from databricks.labs.dqx.profiler.generator import DQGenerator
+
+# Example: Generate is_unique rule from LLM-detected primary key
+detected_pk_columns = ["customer_id", "order_id"]  # Example detected PK
+confidence = "high"
+reasoning = "LLM analysis indicates these columns form a composite primary key based on schema patterns"
+
+# Generate is_unique rule using the detected primary key
+is_unique_rule = DQGenerator.dq_generate_is_unique(
+    column=",".join(detected_pk_columns),
+    level="error",
+    columns=detected_pk_columns,
+    confidence=confidence,
+    reasoning=reasoning,
+    llm_detected=True,
+    nulls_distinct=True  # Default behavior: NULLs are treated as distinct
+)
+
+print("Generated is_unique rule:")
+print(f"Rule name: {is_unique_rule['name']}")
+print(f"Function: {is_unique_rule['check']['function']}")
+print(f"Columns: {is_unique_rule['check']['arguments']['columns']}")
+print(f"Criticality: {is_unique_rule['criticality']}")
+print(f"LLM detected: {is_unique_rule['user_metadata']['llm_based_detection']}")
+print(f"Confidence: {is_unique_rule['user_metadata']['pk_detection_confidence']}")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Integrated Workflow: LLM Detection + Rule Generation
+# MAGIC 
+# MAGIC Here's how to combine LLM-based primary key detection with automatic rule generation:
```

Comment:
Remove this example. Use should not use external methods like `dq_generate_is_unique`